app/controllers/admin/api_users_controller.rb

```diff
@@ -32,7 +32,10 @@ def edit;
     end
 
     def update
-      params[:api_user].delete(:password) if params[:api_user][:password].blank?
+      if params[:api_user][:plain_text_password].blank?
```

Comment:
This is error-prone. if `[:api_user]` is nil, you get a cryptic error:

`NoMethodError: undefined method '[]' for nil:NilClass`

Preferred solution would be to use [Hash#dig](https://ruby-doc.org/core-2.3.0_preview1/Hash.html#method-i-dig). It would return `nil` then:

```ruby
2.3.7 :005 > hash = Hash.new
 => {}
2.3.7 :006 > hash[:api_user][:password]
NoMethodError: undefined method `[]' for nil:NilClass
2.3.7 :007 > hash.dig(:api_user, :password)
 => nil
```